### PR TITLE
chore: update pre-commit

### PR DESCRIPTION
**Pull request by cpcloud**
This PR updates `pre-commit` to use the latest isort and black.

Generally not a fan of these kinds of changes, but we have to update these
dependencies at some point.

Going forward, we should enable dependabot after we move to poetry, which will
automate this process and should result in smaller sets of changes.

Supercedes #2972.

### CODE REVIEW

**Comment by cpcloud** [comment]
How do people feel about batching a run of [`flynt`](https://github.com/ikamensh/flynt#flynt---string-formatting-converter) with this PR to avoid two PRs with a lot of formatting changes?

**Comment by jreback** [comment]
> How do people feel about batching a run of [`flynt`](https://github.com/ikamensh/flynt#flynt---string-formatting-converter) with this PR to avoid two PRs with a lot of formatting changes?

no objection (there is also the pyupgrade set of actions which do this)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
actually this PR is fine.

File: ibis/backends/base/__init__.py [hunk 1]
```diff
@@ -155,8 +155,7 @@ def list_tables(self, like: str = None, database: str = None) -> List[str]:
 
     # @abc.abstractmethod
     def table(self, name: str, database: str = None) -> ir.TableExpr:
-        """
-        """
+        """ """
```
Comment (jreback) [inline ibis/backends/base/__init__.py:158]: should prob just kill these (can be another PR)

**Comment by jreback** [comment]
minor comment for followup

File: ibis/backends/base/__init__.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/base/__init__.py:158]: Yup.
